src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -461,6 +461,12 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 			return prompt;
 		}
 
+		//allow-any-unicode-next-line
+		const customMatch = lineText.match(/.*❯(?=[^❯]*$)/g);
+		if (customMatch) {
+			return customMatch[0];
+		}
```

Comment:
Let's also catch the space at the end, similar to how it's done for pwsh, this will avoid us thinking there's a space at the start of the command:

![image](https://github.com/microsoft/vscode/assets/2193314/eba3420d-8286-4c11-8726-44dd26c7d237)